morph/morphemes.py

```diff
@@ -441,3 +446,105 @@ def analyze2str(self):  # m Str
+
+def transcode_item_pair(el):
+    # this includes the 
+    item = el[1]
+    return (el[0],)+ transcode_item(item)
+
+def transcode_location(loc):
+    return (loc.noteId, loc.fieldName, loc.fieldValue, loc.maturity, loc.guid, loc.weight)
+    
+def save_db_all_morphs(cur, db):
+    # save a morphman db as a table in database
+    tname = 'morphs'
+    fields = "morphid, norm, base, inflected, read, pos, subpos"
+
+    drop_table(cur, tname);
+
+    create_table(cur, tname,fields, ", primary key (morphid)");
+
+    # convert the info in the db into list of tuples
+    tuples = map(transcode_item_pair, enumerate(db.keys()))
+
+    # insert them all at once
+    cur.executemany("INSERT INTO %s (%s) VALUES(?,?,?,?,?,?,?);"%(tname,fields), tuples)
+
+def read_db_all_morphs(cur):
+    # read the morphs as a dictionary, where the key is the morph tuple and
+    # the value is the morphid
+    cur.execute("SELECT * FROM morphs;")
+    rows = cur.fetchall()
+    forDict = map(lambda x: (x[1:], x[0]), rows)
+    return dict(forDict)
+    
+def save_db_locations(cur, db, tname):
+    # save a morphman db as a table in database
+    drop_table(cur, tname);
+    fields = "morphid, noteid, field, fieldvalue, mature, guid, weight"
+    create_table(cur, tname,fields,
+       ", primary key (morphid, noteid, field), foreign key (morphid) references morphs");
+
+    # we need to know the morphsid of each morph
+    morphs = read_db_all_morphs(cur)
+
+    # replace the morph with its morphid
+    # push the morphid as part of the list of locations
+    tuples =map(lambda x: # this is a pair of morph and list of locations
+               list(map(lambda y: (morphs[transcode_item(x[0])],)+transcode_location(y),x[1])),
+               db.items())
+
+    # flatten the list... because we have a list of lists (one list per morph)
+    tuples2 = [val for sublist in tuples for val in sublist]
+
+    cur.executemany("INSERT INTO %s (%s) VALUES(?,?,?,?,?,?,?);"%(tname,fields), tuples2)
+        
+
+def save_db(db, path):
+    # assume that the directory is already created...
+    # exceptions will handle the errors
+
+    # we need to wedge this code in here, while we refactor the code...
+    # morphman stores info in a bunch of files.
+    #
+    # first let us try to create a database with each "file" as a table
+    # so let use the basefilename of the relation as
+    tname = os.path.basename(path)
+    dirName = os.path.dirname(path)
+    # it ends with .db so cut it
+    tname = tname[:-3]
+    dbName = dirName + '/morphman.sqlite'
+
+    conn = connect_db(dbName)
+    with conn:
+        cur = conn.cursor()
+        # fortunately morphman saves "all" first
+        # we only need to save this database, the others
+        # are all subsets of it
+        if (tname == 'all'):
+            save_db_all_morphs(cur, db)
+            tname = 'locations'
+            save_db_locations(cur, db, tname)
```

Comment:
Are you sure you want to overwrite 'tname'?

Why not just use: save_db_locations(cur, db, 'locations') ?